examples/red_teaming/redteaming_simple_calculator/DEFENSE_MIDDLEWARE_README.md

```
@@ -0,0 +1,267 @@
+Can log warnings, block execution, or auto-correct with the correct answer. The prompt structure is built-in and only requires a `tool_description` in the configuration.
+
+### Content Safety Guard (`content_safety_guard`)
+Uses safety guard models (e.g., NVIDIA Nemoguard, Qwen Guard) to detect harmful or unsafe content in function outputs. The middleware analyzes content independently and extracts safety categories when unsafe content is detected. Can log warnings, block execution, or return a user-friendly refusal message. Supports various guard models via NAT's LLM system (NIM, Hugging Face, etc.).
+
+**Note**: To use Hugging Face guard models (for example, Qwen Guard), install the Hugging Face optional dependencies:
+```bash
+pip install "nvidia-nat[huggingface]"
+```
+
+### PII Defense (`pii_defense`)
+Uses Microsoft Presidio (https://github.com/microsoft/presidio) to detect and anonymize Personally Identifiable Information (PII) in function outputs. Configurable with specific entity types and score thresholds. Uses rule-based entity recognition (no LLM required).
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)